packages/editor-ui/src/stores/workflows.store.ts

```diff
@@ -553,11 +554,26 @@ export const useWorkflowsStore = defineStore(STORES.WORKFLOWS, () => {
 	}
 
 	function addExecutingNode(nodeName: string) {
+		clearNodeExecutionQueue();
 		executingNode.value.push(nodeName);
 	}
 
 	function removeExecutingNode(nodeName: string) {
-		executingNode.value = executingNode.value.filter((name) => name !== nodeName);
+		executingNodeCompletionQueue.value.push(nodeName);
+		clearNodeExecutionQueue(true);
+	}
+
+	function clearNodeExecutionQueue(keepLastInQueue = false) {
+		const shouldKeepLastExecutingNode =
+			keepLastInQueue && executingNode.value.length <= executingNodeCompletionQueue.value.length;
+		const lastExecutingNode = executingNodeCompletionQueue.value.at(-1);
+		const nodesToRemove = shouldKeepLastExecutingNode
+			? executingNodeCompletionQueue.value.slice(0, -1)
+			: executingNodeCompletionQueue.value;
+
+		executingNode.value = executingNode.value.filter((name) => !nodesToRemove.includes(name));
+		executingNodeCompletionQueue.value =
+			shouldKeepLastExecutingNode && lastExecutingNode ? [lastExecutingNode] : [];
```

Comment:
I stared at this code for 5mins and couldn't make sense what is happening here. Could we extract the state of `executingNode`, `executingNodeCompletionQueue` and their functions to a separate composable, since `executingNodeCompletionQueue` seems to be internal state to that? That way it would be encapsulated and could have documentation about what it does. Especially since this store is getting bloated